purchase_manual_service_quantity/models/purchase_order_line.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+
+    _inherit = 'purchase.order.line'
+
+    @api.depends('order_id.state', 'move_ids.state')
+    def _compute_qty_received(self):
+        for line in self:
+            if line.product_id.type in ['service'] and\
+                    line.product_id.purchase_manual_received_qty:
+                line.qty_received = line.qty_received_manually
+            else:
+                super(PurchaseOrderLine, self)._compute_qty_received()
+
+    def _inverse_qty_received(self):
+        for line in self:
+            if line.product_id.type in ['service'] and \
+                    line.product_id.purchase_manual_received_qty:
+                line.qty_received_manually = line.qty_received
+
+    qty_received_manually = fields.Float(string="Manually Received Qty",
```

Comment:
Please put fields definition on top, see https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#models

Reply:
Are these fields require decimal precision ?

Reply:
I don't think it's necessary here.